backbone/src/main/java/org/researchstack/backbone/ui/step/body/DateQuestionBody.java

```diff
@@ -142,63 +148,83 @@ public BodyAnswer getBodyAnswerState() {
         return format.validateAnswer(calendar.getTime());
     }
 
-    private void showDialog(TextView tv) {
+    private void showDialog(ViewGroup parent, TextView tv) {
         // need to find a material date picker, since it's not in the support library
-        ContextThemeWrapper contextWrapper = new ContextThemeWrapper(tv.getContext(),
-                R.style.Theme_Backbone);
         if (format.getStyle() == AnswerFormat.DateAnswerStyle.Date) {
-            new DatePickerDialog(contextWrapper,
-                    (view, year, monthOfYear, dayOfMonth) -> {
-                        calendar.set(year, monthOfYear, dayOfMonth);
-                        hasChosenDate = true;
-
-                        // Set result to our edit text
-                        String formattedResult = createFormattedResult();
-                        tv.setText(formattedResult);
-                    },
-                    calendar.get(Calendar.YEAR),
-                    calendar.get(Calendar.MONTH),
-                    calendar.get(Calendar.DAY_OF_MONTH)).show();
+            showDatePicker(parent, tv);
         } else if (format.getStyle() == AnswerFormat.DateAnswerStyle.TimeOfDay) {
-            new TimePickerDialog(contextWrapper,
-                    (view, hourOfDay, minute) -> {
-                        calendar.set(Calendar.HOUR_OF_DAY, hourOfDay);
-                        calendar.set(Calendar.MINUTE, minute);
+            showTimePicker(tv);
+        } else if (format.getStyle() == AnswerFormat.DateAnswerStyle.DateAndTime) {
+            showDatePicker(parent, tv);
+        } else {
+            throw new RuntimeException("DateAnswerStyle " + format.getStyle() + " is not recognised");
+        }
+    }
+
+
+    private void showDatePicker(ViewGroup parent, TextView tv) {
```

Comment:
Why are you passing a ViewGroup since you're only using it to get Context?

Reply:
You can get the context directly from the textView